### PR TITLE
Standardize some date formats

### PR DESCRIPTION
**Pull request by dannyroberts**
This PR mostly has no user-facing changes and is just using constants to replace commonly used date/time formatting strings, but I took liberties with some inconsistent date formats when those were just being used to present to the user. Specifically:
- I changed date format for billing from 08 April 2015 (and a couple other stray ones that were probably different only inadvertently) to Apr 08, 2015 (a format used in a few other places)

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 5 errors:

* corehq/apps/app_manager/management/commands/upload_app_forms.py, line 62 - E501 line too long (135 > 115 characters)
* corehq/apps/reminders/views.py, line 1316 - E203 whitespace before ':'
* corehq/apps/reports/standard/inspect.py, line 212 - E501 line too long (119 > 115 characters)
* custom/_legacy/pathindia/reports.py, line 106 - E501 line too long (116 > 115 characters)
* custom/_legacy/pathindia/reports.py, line 107 - E501 line too long (125 > 115 characters)

**Comment by dannyroberts** [comment]
@nickpell as we discussed @biyeun FYI if you want to push back @NoahCarnahan buddy

**Comment by dannyroberts** [comment]
:white_check_mark:

File: corehq/const.py [hunk 1]
```diff
@@ -1,3 +1,7 @@
 USER_DATETIME_FORMAT = "%b %d, %Y %H:%M %Z"
 USER_DATE_FORMAT = "%b %d, %Y"
 USER_TIME_FORMAT = "%H:%M %Z"
+USER_MONTH_FORMAT = "%B %Y"
+
+SERVER_DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S"
+SERVER_DATETIME_FORMAT_NO_SEC = "%Y-%m-%d %H:%M"
```
Comment (esoergel) [inline corehq/const.py]: :+1: much more readable

**Pull request merged**